python/tidy/servo_tidy/tidy.py

```diff
@@ -694,6 +664,81 @@ def check_spec(file_name, lines):
                 brace_count -= 1
 
 
+def check_config_file(config_file, print_text=True):
+    # Load configs from servo-tidy.toml
+    if os.path.exists(config_file):
+        with open(config_file) as content:
+            conf_file = content.read()
+            lines = conf_file.splitlines(True)
+    else:
+        print("%s config file is required but was not found" % config_file)
+        sys.exit(1)
+
+    if print_text:
+        print '\rChecking for config file...'
+
+    current_table = ""
+    for idx, line in enumerate(lines):
+        # Ignore comment lines
+        if line.strip().startswith("#"):
+            continue
+
+        # Check for invalid tables
+        if re.match("\[(.*?)\]", line.strip()):
+            table_name = re.findall(r"\[(.*?)\]", line)[0].strip()
+            if table_name not in ("configs", "ignore"):
+                yield config_file, idx + 1, "invalid config table [%s]" % table_name
+            current_table = table_name
+            continue
+
+        # Skip if there is no equel sign in line, assuming is not a key
+        if "=" not in line:
+            continue
+
+        key = line.split("=")
+        key = key[0].strip()
+
+        # Check for invalid keys inside [configs] and [ignore] table
+        if (current_table == "configs" and key not in config or
+                current_table == "ignore" and key not in config["ignore"] or
+                # Any key outside of tables
+                current_table == ""):
+            yield config_file, idx + 1, "invalid config key '%s'" % key
+
+    # Skip parsing config file for tests
+    if config_file == CONFIG_FILE_PATH:
+        parse_config(conf_file)
```

Comment:
My concern is that we shouldn't introduce code in tidy just for the sake of tests. It'd be a big win if we avoid it as far as possible. In this case, since `check_config_file` is the only thing that can throw an error, maybe import that into the testsuite and check it instead of calling `collect_errors_for_files`? Then, let's move this `parse_config` thing to `scan` :)